pandas/tests/typing/valid/test_frame.py

```diff
@@ -0,0 +1,834 @@
+    df.to_html(col_space={"col1": 1, "col2": 3})
+
+
+def test_types_resample() -> None:
+    df = pd.DataFrame({"values": [2, 11, 3, 13, 14, 18, 17, 19]})
+    df["date"] = pd.date_range("01/01/2018", periods=8, freq="W")
+    df.resample("M", on="date")
+    # origin and offset params added in 1.1.0
+    # https://pandas.pydata.org/docs/whatsnew/v1.1.0.html
+    df.resample("20min", origin="epoch", offset=pd.Timedelta(2, "minutes"), on="date")
+
+
+def test_types_from_dict() -> None:
+    pd.DataFrame.from_dict({"col_1": [3, 2, 1, 0], "col_2": ["a", "b", "c", "d"]})
+    pd.DataFrame.from_dict({1: [3, 2, 1, 0], 2: ["a", "b", "c", "d"]})
+    pd.DataFrame.from_dict({"a": {1: 2}, "b": {3: 4, 1: 4}}, orient="index")
+    pd.DataFrame.from_dict({"a": {"row1": 2}, "b": {"row2": 4, "row1": 4}})
+    pd.DataFrame.from_dict({"a": (1, 2, 3), "b": (2, 4, 5)})
+    pd.DataFrame.from_dict(
+        data={"col_1": {"a": 1}, "col_2": {"a": 1, "b": 2}}, orient="columns"
+    )
+
+
+@td.skip_if_no("jinja")  # type: ignore[misc]
+def test_pipe() -> None:
+    def foo(df: pd.DataFrame) -> pd.DataFrame:
+        return df
+
+    df1: pd.DataFrame = pd.DataFrame({"a": [1]}).pipe(foo)
+
+    df2: pd.DataFrame = (
+        pd.DataFrame(
+            {
+                "price": [10, 11, 9, 13, 14, 18, 17, 19],
+                "volume": [50, 60, 40, 100, 50, 100, 40, 50],
+            }
+        )
+        .assign(week_starting=pd.date_range("01/01/2018", periods=8, freq="W"))
+        .resample("M", on="week_starting")
+        .pipe(foo)
+    )
+
+    df3: pd.DataFrame = pd.DataFrame({"a": [1], "b": [1]}).groupby("a").pipe(foo)
+
+    df4: pd.DataFrame = pd.DataFrame({"a": [1], "b": [1]}).style.pipe(foo)
+
+
+# set_flags() method added in 1.2.0
+# https://pandas.pydata.org/docs/whatsnew/v1.2.0.html
+def test_types_set_flags() -> None:
+    pd.DataFrame([[1, 2], [8, 9]], columns=["A", "B"]).set_flags(
+        allows_duplicate_labels=False
+    )
+    pd.DataFrame([[1, 2], [8, 9]], columns=["A", "A"]).set_flags(
+        allows_duplicate_labels=True
+    )
+    pd.DataFrame([[1, 2], [8, 9]], columns=["A", "A"])
+
+
+# error: Untyped decorator makes function "test_types_to_parquet" untyped
+@td.skip_if_no("pyarrow")  # type: ignore[misc]
+def test_types_to_parquet() -> None:
+    df = pd.DataFrame([[1, 2], [8, 9]], columns=["A", "B"]).set_flags(
+        allows_duplicate_labels=False
+    )
+    with tm.ensure_clean() as path:
+        df.to_parquet(Path(cast(str, path)))
+    # to_parquet() returns bytes when no path given since 1.2.0
+    # https://pandas.pydata.org/docs/whatsnew/v1.2.0.html
+    # error: Incompatible types in assignment (expression has type "Optional[bytes]",
+    # variable has type "bytes")
+    b: bytes = df.to_parquet()  # type: ignore[assignment]
+
+
+def test_types_to_latex() -> None:
+    df = pd.DataFrame([[1, 2], [8, 9]], columns=["A", "B"])
+    df.style.to_latex(label="some_label", caption="some_caption")
```

Comment:
Hopefully, the last import issue: needs to be skipped when `jinja2` isn't installed.